File: system/multitier/backend-dotnet/Core/Dtos/ViewOrderDetailsResponse.cs
using MyCompany.MyShop.Backend.Core.Entities;

namespace MyCompany.MyShop.Backend.Core.Dtos;

public class ViewOrderDetailsResponse
{
    public string OrderNumber { get; set; } = null!;
    public DateTime OrderTimestamp { get; set; }
    public string Sku { get; set; } = null!;
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal BasePrice { get; set; }
    public decimal DiscountRate { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal SubtotalPrice { get; set; }
    public decimal TaxRate { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal TotalPrice { get; set; }
    public OrderStatus Status { get; set; }
    public string Country { get; set; } = null!;
    public string? AppliedCouponCode { get; set; }
}

File: system/multitier/backend-java/src/main/java/com/mycompany/myshop/backend/core/dtos/ViewOrderDetailsResponse.java
package com.mycompany.myshop.backend.core.dtos;

import com.mycompany.myshop.backend.core.entities.OrderStatus;
import lombok.Data;

import java.math.BigDecimal;
import java.time.Instant;

@Data
public class ViewOrderDetailsResponse {
    private String orderNumber;
    private Instant orderTimestamp;
    private String sku;
    private int quantity;
    private BigDecimal unitPrice;
    private BigDecimal basePrice;
    private BigDecimal discountRate;
    private BigDecimal discountAmount;
    private BigDecimal subtotalPrice;
    private BigDecimal taxRate;
    private BigDecimal taxAmount;
    private BigDecimal totalPrice;
    private OrderStatus status;
    private String country;
    private String appliedCouponCode;
}
